core/src/main/java/com/microsoft/applicationinsights/TelemetryChannel.java

```diff
@@ -39,18 +44,22 @@
 
     TelemetryChannel(URL endpoint) {
         List<HttpPipelinePolicy> policies = new ArrayList<>();
-        HttpClient client = HttpClient.createDefault();
-        HttpPipelineBuilder pipeline = new HttpPipelineBuilder()
+        HttpClient client = LazyAzureHttpClient.getInstance();
+        HttpPipelineBuilder pipelineBuilder = new HttpPipelineBuilder()
                 .httpClient(client);
+        // Add Azure monitor redirect policy to be able to handle v2.1/track redirects
+        policies.add(new AzureMonitorRedirectPolicy());
         // Retry policy for failed requests
         policies.add(new RetryPolicy());
         // TODO handle authentication exceptions
         HttpPipelinePolicy authenticationPolicy = AadAuthentication.getInstance().getAuthenticationPolicy();
         if (authenticationPolicy != null) {
             policies.add(authenticationPolicy);
         }
-        pipeline.policies(policies.toArray(new HttpPipelinePolicy[0]));
-        this.pipeline = pipeline.build();
+        // Add Logging Policy
+        policies.add(new HttpLoggingPolicy(new HttpLogOptions()));
```

Comment:
is this for debugging?

Reply:
yes this will help customers debug and will show any exceptions thrown in the httppipeline

Reply:
how do customers enable it?

Reply:
through AZURE_LOG_LEVEL env variable.

Reply:
have you found this helpful for debugging? if so let's keep it but not put it in public facing docs, and add comment in code explaining how to enable it